PurchasesCoreSwift/Public/Error Handling/ErrorDetails.swift

```diff
@@ -0,0 +1,16 @@
+//
+//  ErrorDetails.swift
+//  PurchasesCoreSwift
+//
+//  Created by Joshua Liebowitz on 7/12/21.
+//  Copyright © 2021 Purchases. All rights reserved.
+//
+
+import Foundation
+
+@objc(RCErrorDetails) public class ErrorDetails: NSObject {
+
+    @objc(RCFinishableKey) public static let finishableKey: NSError.UserInfoKey = "finishable"
+    @objc(RCReadableErrorCodeKey) public static let readableErrorCodeKey: NSError.UserInfoKey = "readable_error_code"
```

Comment:
I guess if we didn't have to keep objc support this would be an enum of `UserInfoKey`s. 
and we _could_ still make it an enum of `UserInfoKeys` and just expose the values as obj-c variables, but we'd just be making it more complex with no real wins. 
Can't think of a nicer way of doing this while preserving compatibility ¯\\_(ツ)_/¯

Reply:
That's right 😢